src/situations/commun/infra/registre_campagne.js

```diff
@@ -0,0 +1,36 @@
+import jQuery from 'jquery';
+import EventEmitter from 'events';
+
+export default class RegistreCampagne extends EventEmitter {
+  constructor ($ = jQuery, urlServeur = process.env.URL_API) {
+    super();
+    this.$ = $;
+    this.urlServeur = urlServeur;
+  }
+
+  recupereCampagne (codeCampagne) {
+    return new Promise((resolve, reject) => {
+      const localCampagne = window.localStorage[this.cleCampagnePourLocalStorage(codeCampagne)];
+
+      if (localCampagne) {
```

Comment:
Pour éviter des régressions je me dis qu'il faudrait mieux ne pas faire comme ça mais plutôt rester sur un `if (this.navigateur.onLine)` comme dans `registreUtilisateur`. En gros je préfèrerait une règle du style "s'il y a du réseau on continue de faire comme avant."

Reply:
Il y a quelque chose que je ne comprend pas.